internal/services/elasticsan/elastic_san_volume_snapshot_data_source_test.go

```diff
@@ -0,0 +1,189 @@
+								SourceId: volumeId.ID(),
+							},
+						},
+					}
+
+					client := clients.ElasticSan.Snapshots
+					if err = client.VolumeSnapshotsCreateThenPoll(ctx, id, snapshot); err != nil {
+						return fmt.Errorf("creating %s: %+v", id, err)
+					}
+
+					return nil
+				}, "azurerm_elastic_san_volume.test"),
+			),
+		},
+		{
+			Config: d.snapshotRestore(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("source_id").IsNotEmpty(),
+				check.That(data.ResourceName).Key("source_volume_size_in_gib").IsNotEmpty(),
+				check.That(data.ResourceName).Key("volume_name").IsNotEmpty(),
+			),
+		},
+		{
+			Config: d.snapshotSource(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				data.CheckWithClientForResource(func(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) error {
+					if _, ok := ctx.Deadline(); !ok {
+						var cancel context.CancelFunc
+						ctx, cancel = context.WithTimeout(ctx, 30*time.Minute)
+						defer cancel()
+					}
+
+					volumeId, err := volumes.ParseVolumeID(state.ID)
+					if err != nil {
+						return err
+					}
+
+					id := snapshots.NewSnapshotID(volumeId.SubscriptionId, volumeId.ResourceGroupName, volumeId.ElasticSanName, volumeId.VolumeGroupName, data.RandomString)
+
+					client := clients.ElasticSan.Snapshots
+					if err = client.VolumeSnapshotsDeleteThenPoll(ctx, id); err != nil {
+						return fmt.Errorf("creating %s: %+v", id, err)
+					}
+
+					return nil
+				}, "azurerm_elastic_san_volume.test"),
+			),
+		},
+	})
+}
+
+func (d ElasticSANVolumeSnapshotDataSource) snapshotSource(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+variable "primary_location" {
+  default = %q
+}
+variable "random_integer" {
+  default = %d
+}
+variable "random_string" {
+  default = %q
+}
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestrg-esvg-${var.random_integer}"
+  location = var.primary_location
+}
+
+resource "azurerm_elastic_san" "test" {
+  name                = "acctestes-${var.random_string}"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  base_size_in_tib    = 1
+  sku {
+    name = "Premium_LRS"
+  }
+}
+
+resource "azurerm_elastic_san_volume_group" "test" {
+  name           = "acctestesvg-${var.random_string}"
+  elastic_san_id = azurerm_elastic_san.test.id
+}
+
+resource "azurerm_elastic_san_volume" "test" {
+  name            = "acctestesv-${var.random_string}"
+  volume_group_id = azurerm_elastic_san_volume_group.test.id
+  size_in_gib     = 1
+}
+`, data.Locations.Primary, data.RandomInteger, data.RandomString)
+}
+
+func (d ElasticSANVolumeSnapshotDataSource) snapshotRestore(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+variable "primary_location" {
+  default = %q
+}
+variable "random_integer" {
+  default = %d
+}
+variable "random_string" {
+  default = %q
+}
```

Comment:
Same for this test config, remove the variables and refer to them using the indexed verbs notation like we do for other hand written data sources/resources in the provider.

Reply:
removed